gtda/homology/simplicial.py

```diff
@@ -1772,3 +1772,179 @@ def plot(Xt, sample=0, homology_dimensions=None, plotly_params=None):
             Xt[sample], homology_dimensions=homology_dimensions,
             plotly_params=plotly_params
             )
+
+
+# @adapt_fit_transform_docs
+class LowerStarFlagPersistence(BaseEstimator, TransformerMixin, PlotterMixin):
+    """TODO"""
+    _hyperparameters = {
+        "homology_dimensions": {
+            "type": (list, tuple),
+            "of": {"type": int, "in": Interval(0, np.inf, closed="left")}
+            },
+        "coeff": {"type": int, "in": Interval(2, np.inf, closed="left")},
+        "extended": {"type": bool},
+        "infinity_values": {"type": (Real, type(None))},
+        "reduced_homology": {"type": bool},
+        "collapse_edges": {"type": bool}
+        }
+
+    def __init__(self, homology_dimensions=(0, 1), coeff=2, extended=True,
+                 infinity_values=np.inf, reduced_homology=False,
+                 collapse_edges=False, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.extended = extended
+        self.infinity_values = infinity_values
+        self.reduced_homology = reduced_homology
+        self.collapse_edges = collapse_edges
+        self.n_jobs = n_jobs
+
+    def _lower_star_extended_diagram(self, X):
+        n_points = max(X.shape)
+        n_points_cone = n_points + 1
+        n_diag = min(X.shape)
+        X = X.tocoo().astype(np.float32)  # ripser needs single precision
+
+        data_diag = np.zeros(n_points_cone, dtype=X.dtype)
+        data_diag[:n_diag] = X.diagonal()
+        max_value = data_diag[:n_diag].max()
+        min_value = data_diag[:n_points].min()
```

Comment:
I think we need to take the minimum over all values (including the implicit zero-vertices), which are in `data_diag[n_points:n_diag]`

Reply:
You are right of course!  And in fact we need to to the same with `max_value` (e.g. for when all values are negative apart from implicit zeros).

Reply:
Ah yeah, true!